src/coreclr/src/jit/hwintrinsicxarch.cpp

```diff
@@ -528,6 +528,118 @@ GenTree* Compiler::impBaseIntrinsic(NamedIntrinsic        intrinsic,
             break;
         }
 
+        case NI_Vector128_AsVector:
```

Comment:
`Vector128<T>` to `Vector<T>` is simple.

If `Vector<T>` is `32-bytes` we need to upcast to `Vector256` (`TYP_SIMD32`)
If `Vector<T>` is `16-bytes` it is a nop as we are already a `TYP_SIMD16`

However, do we need to update the underlying `struct` handle at all since they differ?

Reply:
Looks like we are already updating the underlying struct handle in `impSIMDPopStack`

Reply:
Ah, no, I need to call the verbose overload and pass in  `, /* expectAddr: */ false, sig->retTypeClass`

Reply:
Fixed.